src/test/java/org/junitpioneer/jupiter/resource/SingleTestMethodWithSourceFileScopedTempDirParameter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.resource;
+
+import static org.junitpioneer.jupiter.resource.Assertions.assertCanAddAndReadTextFile;
+import static org.junitpioneer.jupiter.resource.Assertions.assertReadableWriteableTemporaryDirectory;
+import static org.junitpioneer.jupiter.resource.Scope.SOURCE_FILE;
+
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+class SingleTestMethodWithSourceFileScopedTempDirParameter {
+
+	static Path recordedPathForImplicit;
+	static Path recordedPathForExplicit;
+
+	static class TestCases {
```

Comment:
I think a comment explaining why this file is written the way it is (i.e. with static inner classes that aren't `@Nested`) would make it easier to grok them.

Reply:
Thanks for the feedback! I agree, so I've moved this file into `TemporaryDirectoryTests.java` to make the association between the two files clearer, and I've added the following comment to every test case:

> 	// This test case class is static so that JUnit 5 doesn't pick it up
>	// automatically but our own tests can still explicitly find it and run it.

See commit https://github.com/junit-pioneer/junit-pioneer/pull/491/commits/436303a3a56615c99c8a286d0f580a4ba6b7c428.

Reply:
When I made this comment...

```java
class SingleTestMethodWithSourceFileScopedTempDirParameter {
	static class TestCases {
		@Nested
		class Implicit {
			// tests...
		}
	}
}

... it was strange to see a top-level class without tests that had an inner class that was not `@Nested` (so it wouldn't get executed) and had a further inner class _with_ tests. After moving the code around, the setup is no longer strange at all - this arrangement of non-`@Nested` inner classes is pervasive throughout our tests, so I removed the comment again. 😬

Reply:
Aaah, now I understand. Thanks for shedding some light on this. I'm more than happy to see that the comments have been removed, then. Thanks!